operator/api/v1alpha1/clusterconfig_types.go

```diff
@@ -23,13 +23,17 @@ import (
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
+type WorkloadPlaneSpec struct {
+}
+
 // ClusterConfigSpec defines the desired state of ClusterConfig
 type ClusterConfigSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	// Foo is an example field of ClusterConfig. Edit clusterconfig_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	// Information about the Workload Plane.
+	// +optional
+	WorkloadPlane WorkloadPlaneSpec `json:"workloadPlane,omitempty"`
```

Comment:
I'm not sure, to me we will keep 1 network for Control Plane and 1 network for Workload Plane, but we may expose Ingresses on other networks/IPs.

Why we would need to change the CR layout, to me with the same layout as Bootstrap config we are fine

Reply:
My point is that the assumptions about what services should use which network could be wrong (e.g. don't use the public-facing workload plane for the overlay network, or don't expose the control-plane-ingress-controller on the control plane network), and this will likely become easier to handle if we don't encode such assumptions in the CR structure.

The fact that BootstrapConfiguration also has these assumptions encoded would also become a problem, IMO.

Anyway, let's not jump to conclusions on this topic, we'll try to think about it more in the future. And changes to the CR schema are possible, so not much to worry about.